src/main/java/frc/robot/subsystems/Revolver.java

```diff
@@ -56,16 +41,27 @@ public void setAll(double drumSpeed, double feedWheelSpeed) {
     setFeedWheel(feedWheelSpeed);
   }
 
+  public void resetReps(){
+    m_drumReps = 0;
+  }
+
   /**
    * Sets the speed of the drum
    * @param speed the speed to run the drum at as a percentage from -1 to 1
    */
   public void setDrum(double speed) {
     m_drumMotor.set(speed);
-    if (m_pdp.getCurrent(RobotMap.PDP.DRUM_PORT) > Constants.REVOLVER_DRUM_CURRENT_THRESHOLD) {
-      m_repsAboveDrum++;
-    } else {
-      resetDrumReps();
+
+    if((speed==0) || (speed!=0 && m_drumMotor.getEncoder().getVelocity()!=0)){
+      resetReps();
+    }
+
+    if (speed > 0 && m_drumMotor.getEncoder().getVelocity()==0.0){
+      m_drumReps++;
+    }
+    if(m_drumReps > Constants.REVOLVER_REPS_THRESHOLD){
+      new UnjamRevolver(this).schedule(false);
+      resetReps();
     }
```

Comment:
Matt is right, it may make sense to move all of this logic to periodic, this would mean you would need to track the motor input somewhere in order to make sure it is non zero when incrementing reps

Reply:
You can just use `m_drumMotor.get()`